hieradata/role/jenkins-slave.yaml

```diff
@@ -0,0 +1,139 @@
+# Useful registry keys: http://www.jasinskionline.com/TechnicalWiki/Useful-Windows-Registry-Keys.ashx?AspxAutoDetectCookieSupport=1
+# Windows registry: http://www.forensicswiki.org/wiki/Windows_Registry
+# Registry Guide for Windows: http://www.pctools.com/guides/registry/
+#
+psick::windows::registry::use_defaults: true
+apsick::windows::registry::keys:
+  'SSL Cipher Suite Order':
+    key: 'HKEY_LOCAL_MACHINE\SOFTWARE\Policies\Microsoft\Cryptography\Configuration\SSL\00010002!Functions'
+    value: 'TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA384_P256,TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA384_P384,TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA256_P256,TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA256_P384,TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA_P256,TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA_P384,TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA_P256,TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA_P384,TLS_RSA_WITH_AES_256_GCM_SHA384,TLS_RSA_WITH_AES_128_GCM_SHA256,TLS_RSA_WITH_AES_256_CBC_SHA256,TLS_RSA_WITH_AES_128_CBC_SHA256,TLS_RSA_WITH_AES_256_CBC_SHA,TLS_RSA_WITH_AES_128_CBC_SHA,TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384_P384,TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256_P256,TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256_P384,TLS_ECDHE_ECDSA_WITH_AES_256_CBC_SHA384_P384,TLS_ECDHE_ECDSA_WITH_AES_128_CBC_SHA256_P256,TLS_ECDHE_ECDSA_WITH_AES_128_CBC_SHA256_P384,TLS_ECDHE_ECDSA_WITH_AES_256_CBC_SHA_P256,TLS_ECDHE_ECDSA_WITH_AES_256_CBC_SHA_P384,TLS_ECDHE_ECDSA_WITH_AES_128_CBC_SHA_P256,TLS_ECDHE_ECDSA_WITH_AES_128_CBC_SHA_P384,TLS_DHE_DSS_WITH_AES_256_CBC_SHA256,TLS_DHE_DSS_WITH_AES_128_CBC_SH'
+    type: 'dword'
+
+psick::puppet::pe_agent::manage_environment: true
+psick::windows::features::use_defaults: true
+psick::windows::features::install:
+  'file-services':
+    ensure: present
+  'storage-services':
+    ensure: present
+  'net-framework-45-core':
+    ensure: present
+  'was':
+    ensure: present
+  'was-net-environment':
+    ensure: present
+  'was-config-apis':
+    ensure: present
+
+psick::windows::services::use_defaults: true
+psick::windows::services::managed:
+  'RemoteRegistry':
+    ensure: 'stopped'
+    enable: 'manual'
+  'Spooler':
+    ensure: 'running'
+    enable: 'true'
+  'TCPIP6':
+    ensure: 'stopped'
+    enable: 'manual'
+
+# Windows firewall settings.
+windows_firewall::profile_state: 'on'
+windows_firewall::in_policy: 'AllowInbound'
+# FIXME: Why commented? Do we need it?
+# windows_firewall::in_policy: 'BlockInbound'
+windows_firewall::out_policy: 'AllowOutbound'
+windows_firewall::rule_key: 'windows_networks'
+# Purge unmanaged firewall rules
+windows_firewall::purge_rules: false
+
+# Testing users
```

Comment:
Where are these users used?

Reply:
On all the nodes, windows or linux, which include the psick::users class

Reply:
We don't currently use this, we use `ubuntu` and `Administrator` users on Linux and Windows respectively. What would be great, would be to somehow link this to the openvpn, so that every user could have an openvpn configuration generated as well as the existing functionality. Obviously not for now, but a future improvement?

Reply:
I think we might need it to create jenkins user on Windows Jenkins-Slave machine

Reply:
But there's no jenkins user in the list.